internal/ethapi/api.go

```diff
@@ -2541,72 +2765,126 @@ func handlePrivateTransaction(ctx context.Context, b Backend, tx *types.Transact
+	if err != nil {
+		return
+	}
+
+	metadata := engine.ExtraMetadata{
+		ACHashes:     affectedCATxHashes,
+		ACMerkleRoot: merkleRoot,
+		PrivacyFlag:  privateTxArgs.PrivacyFlag,
+	}
+	_, _, data, err = private.P.SendSignedTx(hash, privateTxArgs.PrivateFor, &metadata)
+	if err != nil {
+		return
+	}
+
+	log.Info("sent raw private signed tx",
 		"data", common.FormatTerminalString(data),
 		"hash", hash,
 		"privatefrom", privateTxArgs.PrivateFrom,
 		"privatefor", privateTxArgs.PrivateFor,
-		"affectedCATxHashes", affectedCATxHashes,
-		"merkleroot", merkleRoot,
-		"privacyflag", privateTxArgs.PrivacyFlag)
+		"affectedCATxHashes", metadata.ACHashes,
+		"merkleroot", metadata.ACHashes,
+		"privacyflag", metadata.PrivacyFlag)
+	return
+}
 
+// Quorum
+func handleNormalPrivateTransaction(ctx context.Context, b Backend, tx *types.Transaction, data []byte, privateTxArgs *PrivateTxArgs, from common.Address) (hash common.EncryptedPayloadHash, err error) {
+	affectedCATxHashes, merkleRoot, err := simulateExecutionForPE(ctx, b, from, tx, privateTxArgs)
+	log.Trace("after simulation", "affectedCATxHashes", affectedCATxHashes, "merkleRoot", merkleRoot, "privacyFlag", privateTxArgs.PrivacyFlag, "error", err)
+	if err != nil {
+		return
+	}
+
+	metadata := engine.ExtraMetadata{
+		ACHashes:     affectedCATxHashes,
+		ACMerkleRoot: merkleRoot,
+		PrivacyFlag:  privateTxArgs.PrivacyFlag,
+	}
+	_, _, hash, err = private.P.Send(data, privateTxArgs.PrivateFrom, privateTxArgs.PrivateFor, &metadata)
+	if err != nil {
+		return
+	}
+
+	log.Info("sent private signed tx",
+		"data", common.FormatTerminalString(data),
+		"hash", hash,
+		"privatefrom", privateTxArgs.PrivateFrom,
+		"privatefor", privateTxArgs.PrivateFor,
+		"affectedCATxHashes", metadata.ACHashes,
+		"merkleroot", metadata.ACHashes,
+		"privacyflag", metadata.PrivacyFlag)
 	return
 }
 
+// (Quorum) createPrivacyMarkerTransaction creates a new privacy marker transaction (PMT) with the given signed privateTx.
+// The private tx is sent only to the privateFor recipients. The resulting PMT's 'to' is the privacy precompile address and its 'data' is the
+// privacy manager hash for the private tx.
+func createPrivacyMarkerTransaction(privateTx *types.Transaction, privateTxArgs *PrivateTxArgs) (*types.Transaction, error) {
+	log.Trace("creating privacy marker transaction", "from", privateTx.From(), "to", privateTx.To())
+
+	data := new(bytes.Buffer)
+	err := json.NewEncoder(data).Encode(privateTx)
+	if err != nil {
+		return nil, err
+	}
+
+	_, _, ptmHash, err := private.P.Send(data.Bytes(), privateTxArgs.PrivateFrom, privateTxArgs.PrivateFor, &engine.ExtraMetadata{})
+	if err != nil {
+		return nil, err
+	}
+
+	pmt := types.NewTransaction(privateTx.Nonce(), common.QuorumPrivacyPrecompileContractAddress(), privateTx.Value(), privateTx.Gas(), privateTx.GasPrice(), ptmHash.Bytes())
```

Comment:
@SatpalSandhu61 When we discussed we said that PMT gas limit should be equal to the `intrinsicGas`. From reading this it seems that the PMT gas limit is actually set to the private tx gas limit. Am I understanding correctly? If yes can we update?

A similar question applies to `GasPrice`

Reply:
I think there was a misunderstanding there. What I was trying to say is that since gas is free on GoQuorum, and we never show how much gas a private transaction actually used, users can set an arbitrarily high gasLimit on the private transaction to avoid revealing any information.